### DISCUSSION TITLE
CODEOWNERS should take source and destination branches into account

### DISCUSSION THREAD

**Discussion by chkp-borisko** [community/community#13622] [Pull Requests]
We have a monorepo with several packages and the following workflow:
Anything that goes into the "main" branch is deployed to production immediately.
Team A has a branch "A-Dev" where they aggregate their changes before they merge to master and deploy, they are the owners of package A.
Team B has a branch "B-Dev" where they aggregate their changes before they merge to master and deploy, they are the owners of package B.
Every now and then team B merges the "main" branch into "B-Dev" but if this merge done via a PR they need the approval of someone from team A.
Also when someone from team A merging from "A-Dev" to "main" every change needs to be re-approved by a code owners although they were PRed already when merging from side branch to "A-Dev".
I would like to have an ability to specify source and target branch for sections of code owners or at least the entire file. 
What do you think?

**Comment by HarryWeppner** [comment]
Running into a similar situation and while the idea of CODEOWNERS is good the current signal-to-noise ratio is pretty bad. Like in your case, teams have opted to stage their changes into their own branches before attempting to merge into a "main" branch.

I _think_ that a backward-compatible feature could be to add a repo-level setting to limit CODEOWNERS to a set of (target) branch rules that need to match before a code owner is added to a PR.

@boris-cp if you agree, why not update the heading of this discussion to clearly mark it as a feature request?

**Reply by chkp-borisko** [reply]
@HarryWeppner Please suggest a different heading. This is my first discussion on this forum so maybe I missed some guidelines about keywords.
